lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -894,17 +894,37 @@ void getParameterTypes(Function *F, SmallVectorImpl<TypedPointerType *> &ArgTys,
   if (!RootNode)
     return;
 
+  // Get the parameter list. If the function is a vararg function, drop the last
+  // parameter.
+  NodeArray Params = RootNode->getParams();
+  if (F->isVarArg()) {
+    bool HasVarArgParam = false;
+    if (!Params.empty()) {
+      if (auto *Name = dyn_cast<NameType>(Params[Params.size() - 1])) {
+        if (stringify(Name) == "...")
+          HasVarArgParam = true;
+      }
+    }
+    if (HasVarArgParam) {
+      Params = NodeArray(Params.begin(), Params.size() - 1);
+    } else {
+      LLVM_DEBUG(dbgs() << "[getParameterTypes] function " << MangledName
+                        << " was expected to have a varargs function type\n");
```

Comment:
```suggestion
                        << " was expected to have a varargs parameter\n");
```

as we're inside the `if (F->isVarArg())` here.

Have you considered making this an `llvm_unreachable` instead of just a debug print?

Reply:
Given that `getParameterTypes` will get called on potential user functions, I'm wary of making a path that causes aborts or `llvm_unreachable` on incorrectly mangled functions. (Hell, one of the things that made me discover this bug was someone generating an incorrect mangled name here).

Reply:
Sounds reasonable to leave it as is then, thanks!